openhcl/hcl_mapper/src/lib.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//! Provides a mapper implementation for the page pool that uses [`MshvVtlLow`].
+
+use anyhow::Context;
+use hcl::ioctl::MshvVtlLow;
+use hvdef::HV_PAGE_SIZE;
+use page_pool_alloc::Mapper;
+use page_pool_alloc::PoolType;
+use sparse_mmap::SparseMapping;
+
+/// A mapper that uses [`MshvVtlLow`] to map pages.
+#[derive(Debug)]
+pub struct HclMapper;
+
+impl Mapper for HclMapper {
+    fn map(
+        &self,
+        base_pfn: u64,
+        size_pages: u64,
+        pool_type: PoolType,
+    ) -> Result<SparseMapping, anyhow::Error> {
+        let len = (size_pages * HV_PAGE_SIZE) as usize;
+        let gpa_fd = MshvVtlLow::new().context("failed to open gpa fd")?;
```

Comment:
I suppose it doesn't really matter, given how rarely this method is called, but it might be better to have a `new` method that gets a handle to the fd, and have `map` use it

Reply:
I think we could just store the fd internally like we do with the test mapper, and have a new method?